catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/EntityResourceTest.java

```diff
@@ -175,13 +198,37 @@ public void setup(TestInfo test) throws URISyntaxException, IOException {
     webhookResourceTest.startWebhookEntitySubscriptions(entityName);
 
     UserResourceTest userResourceTest = new UserResourceTest();
-    USER1 = UserResourceTest.createUser(userResourceTest.create(test), authHeaders("test@open-metadata.org"));
+    USER1 = UserResourceTest.createUser(userResourceTest.create(test), adminAuthHeaders());
     USER_OWNER1 = new EntityReference().withId(USER1.getId()).withType("user");
 
+    RoleResourceTest roleResourceTest = new RoleResourceTest();
+    DATA_STEWARD_ROLE =
+        RoleResourceTest.createRole(roleResourceTest.create(DATA_STEWARD_ROLE_NAME), adminAuthHeaders());
+    DATA_STEWARD_ROLE_REFERENCE = new EntityReference().withId(DATA_STEWARD_ROLE.getId()).withType("role");
+    USER_WITH_DATA_STEWARD_ROLE =
+        UserResourceTest.createUser(
+            userResourceTest.create("user-data-steward").withRoles(List.of(DATA_STEWARD_ROLE.getId())),
+            adminAuthHeaders());
+    DATA_CONSUMER_ROLE =
+        RoleResourceTest.createRole(roleResourceTest.create(DATA_CONSUMER_ROLE_NAME), adminAuthHeaders());
+    DATA_CONSUMER_ROLE_REFERENCE = new EntityReference().withId(DATA_CONSUMER_ROLE.getId()).withType("role");
+    USER_WITH_DATA_CONSUMER_ROLE =
+        UserResourceTest.createUser(
+            userResourceTest.create("user-data-consumer").withRoles(List.of(DATA_CONSUMER_ROLE.getId())),
+            adminAuthHeaders());
+
     TeamResourceTest teamResourceTest = new TeamResourceTest();
     TEAM1 = TeamResourceTest.createTeam(teamResourceTest.create(test), adminAuthHeaders());
     TEAM_OWNER1 = new EntityReference().withId(TEAM1.getId()).withType("team");
 
+    PolicyResourceTest.createPolicy(createAccessControlPolicies(), adminAuthHeaders());
+    // Ensure that DefaultAuthorizer gets enough time to load policies before running tests.
+    try {
+      Thread.sleep(8000);
```

Comment:
any reason to introduce this sleep? assuming this is to kick off the thread to update the policies.
IMO it would be better to invoke the policy update through a method call instead of artificial sleep especially in tests. This can cause issues in different systems as the sleep timeout might not be consistent

Reply:
> any reason to introduce this sleep? assuming this is to kick off the thread to update the policies.

Yes
With current implementation, policies are loaded 5 seconds after application starts, and refreshed every 15 minutes
The 8 seconds here gives enough time for the policies to load.

Sleep time in tests usually turn out to be nasty - ends up sending devs into a rabbit hole of chasing race conditions in tests.


There are a few things that I deferred fixing in this PR (since it was getting larger):

1. Replace scheduled policy update in Authorizer with Change Event based update 
2. Introduce a REST API with GET to return all the policies that are loaded in-memory of Authorizer
3. Replace the sleep here using wait until result of the REST API GET call is as expected (with timeout)

I haven't thought through all the details to make the above 3 work, but I have some idea on what to do.

Reply:
#2229